set.mm

```diff
@@ -652048,6 +652049,301 @@ of all sets ( ~ inex1g ).
+  </HTML>
+
+$)
+
+  $( Not before: recld $)
+
+  ${
+    sqrtcvallem1.1 $e |- ( ph -> A e. CC ) $.
+
+    $d A x $.
+    $( Two ways of saying a complex number does not lie on the positive real
+       axis.  Lemma for ~ sqrtcval .  (Contributed by RP, 17-May-2024.) $)
+    sqrtcvallem1 $p |- ( ph -> ( ( ( Im ` A ) = 0 -> ( Re ` A ) <_ 0 ) <->
+                                    -. A e. RR+ ) ) $=
+      ( vx cc crp wcel wn wa cfv cc0 wceq cre wbr wb a1i cr crab clt wo cim cle
+      cdif wi eldif cv cun imor biantrurd reim0b syl notbid eleq1 elrab 3bitr4d
+      bicomd recld 0red lenltd fveq2 breq2d orbi12d syl5bb elun ianor elrp rere
+      bicomi pm5.32i bitri xchbinxr rabbii unrab dfdif2 3eqtr4i eleq2d 3bitr2d
+      ) ABEFUCZGZBEGZBFGHZIZBUAJKLZBMJZKUBNZUDZWAVSWBOABEFUEPAWFBDUFZQGZHZDERZG
+      ZBKWGMJZSNZHZDERZGZTZBWJWOUGZGZVSWFWCHZWETAWQWCWEUHAWTWKWEWPABQGZHZVTXBIZ
+      WTWKAVTXBCUIAXBWTAXAWCAVTXAWCOCBUJUKULUPWKXCOAWIXBDBEWGBLZWHXAWGBQUMULUNP
+      UOAKWDSNZHZVTXFIZWEWPAVTXFCUIAWDKABCUQAURUSWPXGOAWNXFDBEXDWMXEXDWLWDKSWGB
+      MUTVAULUNPUOVBVCWSWQOABWJWOVDPAWRVRBWRVRLAWIWNTZDERWGFGZHZDERWRVRXHXJDEXH
+      WHWMIZXIXKHXHWHWMVEVHXIWHKWGSNZIXKWGVFWHXLWMWHWMXLWHWLWGKSWGVGVAUPVIVJVKV
+      LWIWNDEVMDEFVNVOPVPVQAVTWACUIUO $.
+  $}
+
+  $( Not before absnid: $)
+
+  $( Alternate expression for the absolute value of a real number.  Lemma for
+     ~ sqrtcval .  (Contributed by RP, 11-May-2024.) $)
+  reabsifneg $p |- ( A e. RR -> ( abs ` A ) = if ( A < 0 , -u A , A ) ) $=
+    ( cr wcel cc0 clt wbr cneg cif cabs cfv wa cle wceq wi ltle mpan2 imdistani
+    0re absnid eqcomd syl wn 0red id lenltd bicomd pm5.32i absid sylbi ifeqda )
+    ABCZADEFZAGZAHAIJZUKULUMAUNUKULKZUNUMUOUKADLFZKUNUMMUKULUPUKDBCULUPNRADOPQA
+    SUATUKULUBZKZUNAURUKDALFZKUNAMUKUQUSUKUSUQUKDAUKUCUKUDUEUFUGAUHUITUJT $.
+
+  $( Alternate expression for the absolute value of a real number.
+     (Contributed by RP, 11-May-2024.) $)
+  reabsifnpos $p |- ( A e. RR -> ( abs ` A ) = if ( A <_ 0 , -u A , A ) ) $=
+    ( cr wcel cc0 cle wbr cneg cif cabs cfv wa absnid eqcomd wn wceq wi 0re clt
+    ltnle ltle sylbird mpan imdistani absid syl ifeqda ) ABCZADEFZAGZAHAIJZUGUH
+    UIAUJUGUHKUJUIALMUGUHNZKZUJAULUGDAEFZKUJAOUGUKUMDBCZUGUKUMPQUNUGKUKDARFUMDA
+    SDATUAUBUCAUDUEMUFM $.
+
+  $( Alternate expression for the absolute value of a real number.
+     (Contributed by RP, 11-May-2024.) $)
+  reabsifpos $p |- ( A e. RR -> ( abs ` A ) = if ( 0 < A , A , -u A ) ) $=
+    ( cr wcel cc0 clt wbr cneg cif cabs cfv wa cle wceq 0re ltle mpan imdistani
+    wi absid eqcomd syl wn id 0red lenltd pm5.32i absnid sylbir ifeqda ) ABCZDA
+    EFZAAGZHAIJZUJUKAULUMUJUKKZUMAUNUJDALFZKUMAMUJUKUODBCUJUKUORNDAOPQASUATUJUK
+    UBZKZUMULUQUJADLFZKUMULMUJURUPUJADUJUCUJUDUEUFAUGUHTUIT $.
+
+  $( Alternate expression for the absolute value of a real number.
+     (Contributed by RP, 11-May-2024.) $)
+  reabsifnneg $p |- ( A e. RR -> ( abs ` A ) = if ( 0 <_ A , A , -u A ) ) $=
```

Comment:
To me, this the one version out of the four that could be moved in a future MR to Main.  Indeed, it seems more natural to leave the argument unchanged when the given condition is true (this rules out the versions with condition A<0 and A<=0), and also to leave the argument unchanged as often as possible (so I prefer A>=0 to A>0).  Equivalently, the largest set on which the absolute value is the identity is the set of nonnegative reals.

Reply:
Edit: can you prove it for extended reals (with $\pm \infty$) ?